### ISSUE TITLE
Travis CI failing due to Georgian locale on node 10.7+

### ISSUE THREAD

**Issue by ashsearle** [moment/moment#4762]
All PRs trigger Travis CI jobs to run on various flavour of node (currently ranging from from node version 4 up to 10.) - e.g. PR #4760

Problems in the Georgian locale (ka.js) cause the test suite to fail on node 10.7+.

This is probably due to a change in node 10.7: [Upgrade to ICU 62.1 (Unicode 11, CLDR 33.1)](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md#2018-07-18-version-1070-current-targos)

The [release notes for ICU 62.1](http://site.icu-project.org/download/62) mention:
> Casing behaves differently for Georgian, and differently for that than for any other script

[Migration notes for Unicode 11](http://www.unicode.org/versions/Unicode11.0.0/#Migration) say:
> This change will have major implications for Georgian implementations, including changes for input methods, fonts, casing, and string matching.

A quick 'fix' for Travis failures may be to simply disable locale tests for Georgian / `ka` locale (similar to the way the Turkish / `tr` locale is disabled.)

A full fix would require tweaking Georgian (and Turkish) locales to ensure they work with different levels of unicode support.  (If the only thing that's affected by the unicode revisions is rules for upper/lower-casing, this should be easy to detect by checking the result of passing an affected character through `toUpperCase()` / `toLowerCase()`.)

**Comment by srl295** [comment]
This is not a node.js bug, but it is a new Unicode version 11 - https://github.com/nodejs/node/issues/22518

**Comment by srl295** [comment]
https://github.com/moment/moment/pull/4828 seems to disable the `ka` locale tests

**Comment by srl295** [comment]
I don't know this codebase, but I'm struggling to understand why `ka` fails but `fr` succeeds in a very similar test. 

```
> 'avril'.toLocaleUpperCase('fr')
'AVRIL'
```

**Comment by srl295** [comment]
<img width="1034" alt="sin titulo 2" src="https://user-images.githubusercontent.com/855219/47310217-51ba6100-d5eb-11e8-8c76-4fd9be050efb.png">

Specifically, I think moment.js tests are failing right here (in the debugger):   because `/^ოქტომბერი|^ოქტ/i.test('ᲝᲥᲢ' /* === 'ოქტ'.toUpperCase() */ )` is false but it should be true.   ICU4C, the underlying lib used for regex here, gets this right. v8 bug??

**Comment by srl295** [comment]
Update update.  The bug is actually in moment.js, needed to use the `u` flag.

`/^ოქტომბერი|^ოქტ/iu.test('ᲝᲥᲢ' /* === 'ოქტ'.toUpperCase() */ )` returns true. 

I'll open a PR.

**Comment by ashsearle** [comment]
@srl295 I've looked at this a little bit, and went down a rabbit hole when I started considering legacy support (e.g. Travis CI runs against various old versions of node and old browsers.)

Detecting and applying unicode flag only where it's supported is relatively easy:

```js
var caseInsensitiveFlags = ('unicode' in /test/) ? 'iu' : 'i';
var someRegex = new RegExp('...', caseInsensitiveFlags);
```

I think you've already found the tests rely on `toUpperCase` / `toLowerCase`, which probably need to use `toLocaleUpperCase` / `toLocaleLowerCase` (preferably using the optional locale argument: so `toLocaleUpperCase('ka')` when handling Georgian.

If you do that across all locales, you then find that the tests for Greek don't work, because round-tripping things to upper-case and back loses accents; so you start looking at `localeCompare` to ignore differences in accents: e.g. `someString.localeCompare(greekDay, 'el', { sensitivity: base })`

But then, you have to look at browser-level support for `localeCompare` (added in IE11) and see that moment docs still state moment supports "IE 8, 9, and 10 on Windows 7", so... you need to handle that.

**Comment by srl295** [comment]
There’s a comment on
https://bugs.chromium.org/p/v8/issues/detail?id=8348#c4 that seems to say u
is not needed

**Comment by srl295** [comment]
@ashsearle ☝️ v8 seems to think it's a v8 bug.

**Comment by ashsearle** [comment]
@srl295 I guess that might fix the Georgian issue (if `/\u10d0/i.test('\u1c90') === true` as well as `/\u10d0/iu.test('\u1c90') === true`)

My other comments about (moment tests) needing to use toLocaleUpper/LowerCase and localeCompare still stand thought.  (Tricky work required to fix moment tests for Turkish and Azerbaijani, without breaking Greek.)

**Comment by marwahaha** [comment]
fixed by #4828. Remaining work in #4839
